### PR TITLE
add dark-mode support

### PR DESCRIPTION
**Pull request by RubenWaterman**
these changes add support for dark-mode in the front-end

### CODE REVIEW

**Comment by riggraz** [comment]
Hey RubenWaterman,

thanks for your PR, the dark theme is really cool!

I tried it on my machine and I'd like to point out some things I noticed:

- Comment's actions like "Reply", "Edit", etc., are still black so they are not visible
- Textboxes' text and background color is black when they don't have focus, so it's difficult to read
- Like button does not change style when hovering the mouse over it or when it's been pressed, so the user doesn't know if it has already pressed it or not
- flash messages look really bad on dark theme
- text boxes become white when they have focus (I don't know if that is intended: maybe it'd be better to keep them dark but to change the text color to white)

Also I don't understand why you commited your logos in the PR (lol). You should of course leave astuto's ones!

Let me know, and thanks again for taking the time to contribute to Astuto!

File: app/javascript/stylesheets/components/Post.scss [hunk 1]
```diff
@@ -118,4 +118,15 @@
       color: $astuto-black;
     }
   }
+}
+
+@media (prefers-color-scheme: dark) {
+
+  .postDescription {
+    color: white !important;
```
Comment (riggraz) [inline app/javascript/stylesheets/components/Post.scss:126]: Are all these `!important` necessary?

Comment (RubenWaterman) [reply app/javascript/stylesheets/components/Post.scss:126]: I'm not entirely sure why sometimes it's necessary and other times it works without the !important. I've deleted them wherever possible.

**Comment by RubenWaterman** [comment]
Sorry about the logo's. This is my first contribution and I wasn't aware that anything I push to my master branch will also go into the pull request.

File: app/javascript/stylesheets/components/Post.scss (hunk 1 above)
Comment (riggraz) [reply app/javascript/stylesheets/components/Post.scss:126]: It's probably because of CSS specificity, I removed some of them with the last commit.

**Comment by riggraz** [comment]
Okay I made some changes, it should look a little better now.

I think that the approach used to implement dark mode is probably not the best: we added new CSS selectors for dark mode, whereas we could've used CSS variables (see [this article](https://medium.com/@mwichary/dark-theme-in-a-day-3518dde2955a)). Basically I think we just needed to define color variables both for light and for dark theme, then we would "activate" the light color variables if theme is light, the dark color variables if theme is dark and all the other CSS would remain the same.

Right now I have no time to refactor the CSS in the way I explained earlier. If you have the time you can do it, otherwise no problem, just let me know and I will merge the PR anyway.

**Comment by RubenWaterman** [comment]
I agree that the color variables is a much cleaner approach. However, having javascript to change the selection of which mode you want to have is then the "ugly" part. Having just CSS that "discovers" whether the page should be displayed in light or dark mode has my preference for my use case. I'm happy with the result as is for our page so I'll leave it at the current state. Thanks anyways for the feedback and improvements and of course for making the whole project! It's awesome!

**Comment by riggraz** [comment]
Okay, thank you again for the PR! I am merging it now!

**Comment by fabioespinosa** [comment]
Hey @riggraz, do you happen to know how can I switch dark mode off?

Thanks!

**Comment by fabioespinosa** [comment]
Nevermind, I found out it is being set by prefers-color-scheme media query.

Would be nice for there to be a toggle. I might want dark theme in many places, yet not sure if in a feedback website.

**Pull request merged**